libminifi/src/ListenHTTP.cpp

```diff
@@ -0,0 +1,395 @@
+
+	if (context->getProperty(SSLCertificate.getName(), sslCertFile) && !sslCertFile.empty())
+	{
+		_logger->log_info("ListenHTTP using %s: %s",
+				SSLCertificate.getName().c_str(),
+				sslCertFile.c_str());
+	}
+
+	// Read further TLS/SSL options only if TLS/SSL usage is implied by virtue of certificate value being set
+	std::string sslCertAuthorityFile;
+	std::string sslVerifyPeer;
+	std::string sslMinVer;
+
+	if (!sslCertFile.empty())
+	{
+		if (context->getProperty(SSLCertificateAuthority.getName(), sslCertAuthorityFile)
+				&& !sslCertAuthorityFile.empty())
+		{
+			_logger->log_info("ListenHTTP using %s: %s",
+					SSLCertificateAuthority.getName().c_str(),
+					sslCertAuthorityFile.c_str());
+		}
+
+		if (context->getProperty(SSLVerifyPeer.getName(), sslVerifyPeer))
+		{
+			if (sslVerifyPeer.empty() || sslVerifyPeer.compare("no") == 0)
+			{
+				_logger->log_info("ListenHTTP will not verify peers");
+			}
+			else
+			{
+				_logger->log_info("ListenHTTP will verify peers");
+			}
+		}
+		else
+		{
+			_logger->log_info("ListenHTTP will not verify peers");
+		}
+
+		if (context->getProperty(SSLMinimumVersion.getName(), sslMinVer))
+		{
+			_logger->log_info("ListenHTTP using %s: %s",
+					SSLMinimumVersion.getName().c_str(),
+					sslMinVer.c_str());
+		}
+	}
+
+	std::string headersAsAttributesPattern;
+
+	if (context->getProperty(HeadersAsAttributesRegex.getName(),headersAsAttributesPattern)
+			&& !headersAsAttributesPattern.empty())
+	{
+		_logger->log_info("ListenHTTP using %s: %s",
+				HeadersAsAttributesRegex.getName().c_str(),
+				headersAsAttributesPattern.c_str());
+	}
+
+	auto numThreads = getMaxConcurrentTasks();
+
+	_logger->log_info("ListenHTTP starting HTTP server on port %s and path %s with %d threads",
+			listeningPort.c_str(),
+			basePath.c_str(),
+			numThreads);
+
+	// Initialize web server
+	std::vector<std::string> options;
+	options.push_back("enable_keep_alive");
+	options.push_back("yes");
+	options.push_back("keep_alive_timeout_ms");
+	options.push_back("15000");
+	options.push_back("num_threads");
+	options.push_back(std::to_string(numThreads));
+
+	if (sslCertFile.empty())
+	{
+		options.push_back("listening_ports");
+		options.push_back(listeningPort);
+	}
+	else
+	{
+		listeningPort += "s";
+		options.push_back("listening_ports");
+		options.push_back(listeningPort);
+
+		options.push_back("ssl_certificate");
+		options.push_back(sslCertFile);
+
+		if (!sslCertAuthorityFile.empty())
+		{
+			options.push_back("ssl_ca_file");
+			options.push_back(sslCertAuthorityFile);
+		}
+
+		if (sslVerifyPeer.empty() || sslVerifyPeer.compare("no") == 0)
+		{
+			options.push_back("ssl_verify_peer");
+			options.push_back("no");
+		}
+		else
+		{
+			options.push_back("ssl_verify_peer");
+			options.push_back("yes");
+		}
+
+		if (sslMinVer.compare("SSL2") == 0)
+		{
+			options.push_back("ssl_protocol_version");
+			options.push_back(std::to_string(0));
+		}
+		else if (sslMinVer.compare("SSL3") == 0)
```

Comment:
@apiri  Are we concerned about POODLE? Do we want to disallow any version < TLS1.2?